operator/v1/types_clustermanager.go

```diff
@@ -108,6 +108,24 @@ type RegistrationHubConfiguration struct {
 	//  	he can set featuregate/Foo=false before upgrading. Let's say the cluster-admin wants featuregate/Foo=false.
 	// +optional
 	FeatureGates []FeatureGate `json:"featureGates,omitempty"`
+
+	// RegistrationDrivers represent the list of hub registration drivers that contain information used by hub to initialize the hub cluster
+	// A RegistrationDriverHub contains details of authentication type and the hub cluster ARN
+	// +optional
+	RegistrationDrivers []RegistrationDriverHub `json:"registrationDrivers,omitempty"`
+}
+
+type RegistrationDriverHub struct {
+
+	// Type of the authentication used by hub to initialize the Hub cluster. Possible values are csr and awsirsa.
+	// +required
+	// +kubebuilder:default:=csr
+	// +kubebuilder:validation:Enum=csr;awsirsa
+	AuthType string `json:"authType,omitempty"`
```

Comment:
one minor comment,  we should add

// +listType=map
// +listMapKey=authType
here, so we will not set duplicate type